modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -0,0 +1,493 @@
+													symbol="search"
+												/>
+											</ClayInput.GroupInsetItem>
+										</ClayInput.GroupItem>
+									</ClayInput.Group>
+								</ManagementToolbar.Item>
+							</ManagementToolbar.ItemList>
+						</ManagementToolbar.Container>
+
+						<div className="fields pb-2 pt-2">
+							{visibleFields.length ? (
+								visibleFields.map(({name, selected}) => (
+									<div
+										className="pb-2 pl-3 pr-3 pt-2"
+										key={name}
+									>
+										<ClayCheckbox
+											checked={selected}
+											label={name}
+											onChange={({target: {checked}}) => {
+												setFields(
+													fields.map((field) =>
+														field.name === name
+															? {
+																	...field,
+																	selected: checked,
+															  }
+															: field
+													)
+												);
+											}}
+										/>
+									</div>
+								))
+							) : (
+								<div className="pb-2 pl-3 pr-3 pt-2 text-3">
+									{Liferay.Language.get('no-results-found')}
+								</div>
+							)}
+						</div>
+					</>
+				)}
+			</ClayModal.Body>
+
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton
+							onClick={() => {
+								closeModal();
+
+								onSave();
+							}}
+						>
+							{Liferay.Language.get('save')}
+						</ClayButton>
+
+						<ClayButton
+							displayType="secondary"
+							onClick={() => closeModal()}
+						>
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+					</ClayButton.Group>
+				}
+			/>
+		</div>
+	);
+};
+
+const Fields = ({fdsView, fdsViewsURL}: FDSViewSectionInterface) => {
+	const [fdsFields, setFDSFields] = useState<Array<FDSFieldType> | null>(
+		null
+	);
+
+	const fdsFieldsOrderRef = useRef('');
+
+	const getFDSFields = async () => {
+		const response = await fetch(
+			`${API_URL.FDS_FIELDS}?filter=(${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_FIELD_ID} eq '${fdsView.id}')&nestedFields=${OBJECT_RELATIONSHIP.FDS_VIEW_FDS_FIELD}`
+		);
+
+		if (!response.ok) {
+			openToast({
+				message: Liferay.Language.get(
+					'your-request-failed-to-complete'
+				),
+				type: 'danger',
+			});
+
+			return null;
+		}
+
+		const responseJSON = await response.json();
+
+		const storedFDSFields = responseJSON?.items;
+
+		if (!storedFDSFields) {
+			openToast({
+				message: Liferay.Language.get(
+					'your-request-failed-to-complete'
+				),
+				type: 'danger',
+			});
+
+			return null;
+		}
+
+		const fdsFieldsOrder =
+			storedFDSFields?.[0]?.[OBJECT_RELATIONSHIP.FDS_VIEW_FDS_FIELD]
+				?.fdsFieldsOrder;
+
+		if (fdsFieldsOrder) {
+			fdsFieldsOrderRef.current = fdsFieldsOrder;
+
+			const storedOrderedFDSFieldIds = fdsFieldsOrder.split(',');
+
+			const orderedFDSFields: Array<FDSFieldType> = [];
+
+			const orderedFDSFieldIds: Array<number> = [];
+
+			storedOrderedFDSFieldIds.forEach((fdsFieldId: string) => {
+				storedFDSFields.forEach((storedFDSField: any) => {
```

Comment:
`any` on purpose?

Reply:
An oversight, we can improve it. Nice catch, there are a good amount of `any`s in this PR that **are** intentional :slightly_smiling_face: